analysis/quantifier-comparisons/pre-processing-benchmarking-metrics.Rmd

```diff
@@ -0,0 +1,612 @@
+Alevin-fry seems to always be in the middle and it looks like the `--sketch` mode leads to increased UMI/cell with the same number of cells (see previous plots looking at the number of cells).
+
+What about genes/cell? Are the same trends consistent?
+
+```{r}
+cell_qc_filter %>%
+  mutate(tool = fct_reorder(tool, detected, .fun = 'mean')) %>%
+  ggplot(aes(x = reorder(tool, detected), y = detected, fill = tool)) + 
+  geom_boxplot() + 
+  theme_classic() + 
+  ylab("Number of Genes detected/Cell") + 
+  xlab("")
+```
+
+
+```{r}
+ggplot(cell_qc_filter, aes(y = detected, fill = tool)) + 
+  geom_boxplot() + 
+  facet_wrap(~sample) + 
+  theme_classic() + 
+  ylab("Genes Detected/Cell") +
+  theme(axis.ticks.x = element_blank(), axis.text.x = element_blank())
+  
+```
+
+So we see a lot more variation in the genes detected/cell across tools than we did in the UMI/cell. Alevin is highest in the same 126 and 127 samples and cellranger consistently seems to be on the low end. Alevin-fry in `--sketch` mode again seems to get more coverage than alevin-fry without `--sketch` mode. 
+
+Previously Josh had found that Kallisto detects more UMI's and more genes/cell but that doesn't always seem to be the case. It looks like in 126 and 127, Alevin and Alevin-fry `--sketch` are higher. But again, cellranger is keeping more cells than the other tools in those 2 samples. 
+
+What if we just look at cells that are found in all of the tools? 
+
+```{r}
+cell_counts <- cell_qc_filter %>%  
+  dplyr::count(cell_id, sample)
+
+common_cells <- cell_counts %>%
+  dplyr::filter(n == 5) %>%
+  dplyr::pull(cell_id)
+
+cell_qc_common <- cell_qc_filter %>%
+  dplyr::filter(
+    (cell_id %in% common_cells) 
+  )
+```
+
+Now what happens to our mito content, UMI/cell, and genes/cell? 
+
+```{r}
+# mito comparison across shared cells only of all 5 runs
+ggplot(cell_qc_common, aes(y = subsets_mito_percent, fill = tool)) + 
+  geom_boxplot() + 
+  facet_wrap(~sample) + 
+  theme_classic() + 
+  ylab("% Mito /Cell") + 
+  theme(axis.ticks.x = element_blank(), axis.text.x = element_blank())
+```
+```{r}
+# number of UMI's/sample in shared cells only 
+ggplot(cell_qc_common, aes(x = tool, y = sum, fill = tool)) + 
+  geom_boxplot() + 
+  facet_wrap(~sample) +
+  theme_classic() + 
+  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
+  ylab("UMI/cell") + 
+  xlab("")
+```
+```{r}
+# genes detected/cell in shared cells only 
+ggplot(cell_qc_common, aes(y = detected, fill = tool)) + 
+  geom_boxplot() + 
+  facet_wrap(~sample) + 
+  theme_classic() + 
+  ylab("Genes Detected/Cell") +
+  theme(axis.ticks.x = element_blank(), axis.text.x = element_blank())
+  
+```
+Wow, so everything is very consistent when you look at only shared cells except for the genes/cell. Now, we actually see that alevin-fry `--sketch` captures the highest number of genes/cell in shared cells only, with cellranger, alevin, and alevin-fry without `--sketch` all capturing around the same. 
```

Comment:
This is very reassuring! I wonder if the extra genes detected with `--sketch` are all low frequency? Can we redo this plot this excluding genes with low counts? It looks like the easiest way to do that is to play with the `threshold` argument to `addPerCellQC()` (or use `perCellQCMetrics()` to get a separate table.)